instrumentation/github.com/gin-gonic/gin/otelgin/gintrace_test.go

```diff
@@ -105,3 +109,54 @@ func TestPropagationWithCustomPropagators(t *testing.T) {
 
 	router.ServeHTTP(w, r)
 }
+
+func TestWithCustomSpanOptions(t *testing.T) {
+	spanRecorder := tracetest.NewSpanRecorder()
+	provider := sdktrace.NewTracerProvider(sdktrace.WithSpanProcessor(spanRecorder))
+
+	r := httptest.NewRequest("GET", "/user/123", nil)
+	w := httptest.NewRecorder()
+
+	// test for overwriting span start/end time, it may be useless
+	spanStartTime := time.Now().Add(time.Second)
+	spanEndTime := spanStartTime.Add(time.Minute)
+
+	startOpts := []trace.SpanStartOption{
+		trace.WithTimestamp(spanStartTime),
+		trace.WithAttributes(
+			[]attribute.KeyValue{
+				attribute.String("foo", "bar"),
+			}...,
+		),
+	}
+
+	endOpts := []trace.SpanEndOption{
+		trace.WithTimestamp(spanEndTime),
+		trace.WithStackTrace(true), // capture the error with stack trace
+	}
+
+	router := gin.New()
+	router.Use(Middleware("foobar", WithTracerProvider(provider), WithSpanStartOptions(startOpts...), WithSpanEndOptions(endOpts...)))
```

Comment:
I think it would be easier to read if there were two tests, one for each option.

Reply:
ok, I change it into two tests, one for span start options, one for span end options